Makefile

```
@@ -0,0 +1,32 @@
+build:
+	poetry build
+
+clean:
+	rm -rf libhoney/__pycache__/
+	rm -rf dist/*
+
+install:
+	poetry install --no-root --no-ansi
+
+lint:
+	poetry run pylint --rcfile=pylint.rc libhoney
+
+format:
+	poetry run pycodestyle libhoney --max-line-length=140
+
+test:
+	poetry run coverage run -m unittest discover -v
+
+smoke:
+	@echo ""
+	@echo "+++ Running example app in docker"
+	@echo ""
+	docker-compose up --build
```

if we make docker-compose fail when the app fails, we'll get a [CI failure](https://app.circleci.com/pipelines/github/honeycombio/libhoney-py/1233/workflows/5fcc8366-b464-4b10-b43e-cae860df8815/jobs/9700)

```suggestion
	docker-compose up --build --exit-code-from factorial-example
```

---

ooh yes good call!